src/libcore/str/mod.rs

```diff
@@ -337,6 +344,7 @@ fn unwrap_or_0(opt: Option<&u8>) -> u8 {
 /// Reads the next code point out of a byte iterator (assuming a
 /// UTF-8-like encoding).
 #[unstable(feature = "core")]
+#[inline]
```

Comment:
This seems like a pretty big function to force an inline hint on, just confirming that this is intentional.